### DISCUSSION TITLE
Fail on building (wrong wiki URL) :)

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [Andrew-Chen-Wang/github-wiki-action#14] [Q&A] [answered]
Hey there,

[I'm getting this error](https://github.com/201flaviosilva/HillClimbing.js/runs/6472581198?check_suite_focus=true):

```sh
Run Andrew-Chen-Wang/github-wiki-action@v3
/usr/bin/docker run --name dd1ff46e74abf427fa3a3f2b4969042dd_5453a0 --label 08450d --workdir /github/workspace --rm -e WIKI_DIR -e GH_TOKEN -e GH_MAIL -e GH_NAME -e EXCLUDED_FILES -e INPUT_WIKI_DIR -e INPUT_GH_TOKEN -e INPUT_GH_MAIL -e INPUT_GH_NAME -e INPUT_REPO -e INPUT_WIKI_PUSH_MESSAGE -e INPUT_EXCLUDED_FILES -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/HillClimbing.js/HillClimbing.js":"/github/workspace" 08450d:d1ff46e74abf427fa3a3f2b4969042dd
REPO ENV is missing. Using the current one.
Cloning into 'temp_wiki_622accf3f238ac9d08144d8eca3d57e616fbde65'...
Cloning wiki git...
Message:
try fix wiki build
Copying files to Wiki
sending incremental file list
./
Home.md

sent 170 bytes  received 38 bytes  416.00 bytes/sec
total size is 56  speedup is 0.27
rm: can't remove 'temp_wiki_622accf3f238ac9d08[14](https://github.com/201flaviosilva/HillClimbing.js/runs/6472581198?check_suite_focus=true#step:4:14)4d8eca3d57e616fbde65/_ignore': No such file or directory
Pushing to Wiki
[master 4df3484] try fix wiki build
 1 file changed, 2 insertions(+)
remote: Permission to [20](https://github.com/201flaviosilva/HillClimbing.js/runs/6472581198?check_suite_focus=true#step:4:20)1flaviosilva/HillClimbing.js.wiki.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/201flaviosilva/HillClimbing.js.wiki.git/': The requested URL returned error: 403
```

For what I understood this is trying to find a GitHub wiki in https://github.com/201flaviosilva/HillClimbing.js.wiki/ but my GitHub wiki is in the https://github.com/201flaviosilva/HillClimbing.js/wiki , any help ? 😅

- Repository: https://github.com/201flaviosilva/HillClimbing.js
- Error: https://github.com/201flaviosilva/HillClimbing.js/runs/6472581198?check_suite_focus=true
- Wiki: https://github.com/201flaviosilva/HillClimbing.js/wiki
- DeployWiki.yml: https://github.com/201flaviosilva/HillClimbing.js/blob/main/.github/workflows/DeployWiki.yml

(Sorry if it was confusing)

**Comment by Andrew-Chen-Wang** [comment, marked as answer]
https://github.com/201flaviosilva/HillClimbing.js.wiki/ is the actual git repository uri.

Two things could be happening:
- You originally had this in private, which I doubt since it's an NPM package. But if you did, you would've needed a GitHub PAT
- More likely, the email you used is not the one associated with your GitHub account, but I could be wrong.

Otherwise, it might be a bug? The new version was submitted by a user that I saw had this working fine, so not entirely sure what else could be happening.

**Reply by 201flaviosilva** [reply]
This git repository has always been a public repository. And the associated email it's the right, i this is probably a bug, but I will try run again with this: `GH_MAIL: ${{ github.event.pusher.email }}`

**Reply by 201flaviosilva** [reply]
Fails again: https://github.com/201flaviosilva/HillClimbing.js/runs/6473512676?check_suite_focus=true

**Reply by Andrew-Chen-Wang** [reply]
cc @rcdailey I'm not completely sure of this issue. If you have the time, do you mind taking a look at this? Thanks!

**Reply by 201flaviosilva** [reply]
Hey it’s me again, if this make sense, I [forked](https://github.com/201flaviosilva-labs/Beep-Wiki-Action-Test) my repository to a my [own organization](https://github.com/201flaviosilva-labs) where I make some tests and etc... and the action works fine there: https://github.com/201flaviosilva-labs/Beep-Wiki-Action-Test/runs/6475732195?check_suite_focus=true

**Reply by rcdailey** [reply]
@Andrew-Chen-Wang I'll peek at this today.

**Reply by rcdailey** [reply]
I'm not seeing anything wrong with the script. The return code is 403 so he has some sort of authentication issue. It says this:

> remote: Permission to 201flaviosilva/HillClimbing.js.wiki.git denied to github-actions[bot]

Why is "github-actions[bot]" the user trying to push the repo? The bot isn't going to have permission to push. When I run my wiki update pipeline, I use my personal email and PAT and it works fine.

**Reply by rcdailey** [reply]
@201flaviosilva I think I see the issue. Look at the logs [here](https://github.com/201flaviosilva/HillClimbing.js/runs/6475324921?check_suite_focus=true#step:1:16):

```txt
GITHUB_TOKEN Permissions
  Contents: read
  Metadata: read
```

You need to give `GITHUB_TOKEN` write permissions, I think. Read more [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).

In my personal project, I use a personal access token (which already has permission to push to my repo) and it works:

```yml
env:
  WIKI_DIR: wiki/
  GH_TOKEN: ${{ secrets.PAT }}
  GH_MAIL: ${{ secrets.EMAIL }}
  GH_NAME: ${{ github.repository_owner }}
  EXCLUDED_FILES: "*.json"
```

**Reply by Andrew-Chen-Wang** [reply]
Previously, the GITHUB_TOKEN should be working fine. Reading this, https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token @201flaviosilva can you try adding the additional permissions using the `permissions` label? If that doesn't work, we may need to revise the README and say that GITHUB_TOKEN doesn't work.

I'll take a look at the community and see whether GITHUB_TOKEN works for anyone using v3.

Thanks for taking a look at this @rcdailey greatly appreciated!

**Reply by rcdailey** [reply]
The only functional difference I can see is that we use `$GH_NAME:$GH_TOKEN@github.com` instead of just `$GH_TOKEN@github.com` in the clone URL. Push used both before. But I'm fairly confident only one should be used. I'm not able to find official documentation from Github that shows which one is proper, but based on my searching (see references below), the latter seems to be the most commonly used format. Should we test with just the GH_TOKEN in the clone URL and see if that makes a difference?

At the end of the day, I don't see how GITHUB_TOKEN should work anyway without being in permissive mode. There's quite simply no documentation that I've found that shows it should work in restrictive mode. I agree sampling the community and further auditing the documentation is a good thing to do, though.

References: [1](https://stackoverflow.com/questions/56052926/git-clone-works-with-personal-access-token-but-not-git-push), [2](https://stackoverflow.com/questions/68234568/how-to-use-github-token-to-clone-a-private-repository), [3](https://github.blog/2012-09-21-easier-builds-and-deployments-using-git-over-https-and-oauth/)

**Reply by 201flaviosilva** [reply]
Sorry for only replying now, I was on my way home.

But finally the problem is solved :)

It was really a permissions issue. Thank you very much for the tool and the excellent support, you saved my life 😁

(And sorry for wasting your time with this silly problem 😅)

<img width="806" alt="Captura de ecrã 2022-05-17, às 21 50 25" src="https://user-images.githubusercontent.com/55815265/168908154-8f3216f6-eb6c-4982-b613-5ad421694f4d.png">

**Reply by Andrew-Chen-Wang** [reply]
No I definitely learned something today as well :) Thanks for asking!
